### ISSUE TITLE
code

### ISSUE THREAD

**Issue by ManXu1** [NickMts/Energy-Efficient-OFDM-with-Intelligent-PAPR-aware-Adaptive-Modulation#2]
Dear author, in the code **J = order.size(dim=1)** leads to a later **arr = y_pred[:, K_d:].reshape(J, K_d)** with incorrect dimensions, and a subsequent **symbol = torch.cat((symbol[:, 0:int(K_d/2)], torch.zeros(1, l *K_d), symbol[:, int(K_d/2):K_d]), dim=-1)** in **torch.zeros(1, l*K_d)**, dimensions don't match. I removed Reshape, 1 in torch.zeros(1, l*K_d) modified to 64 and got loss as Epoch: 0, Batch: 0, Loss: inf. Please check out the code, thanks for sharing the code. I'm looking forward to hearing back from you.
